apps/trading/e2e/tests/wallet/test_wallet.py

```diff
@@ -0,0 +1,116 @@
+from fixtures.market import setup_continuous_market
+
+order_size = "order-size"
+order_price = "order-price"
+place_order = "place-order"
+order_side_sell = "order-side-SIDE_SELL"
+market_order = "order-type-Market"
+tif = "order-tif"
+expire = "expire"
+
+
+@pytest.fixture(scope="module")
+def vega(request):
+    with init_vega(request) as vega:
+        yield vega
+
+
+@pytest.fixture(scope="module")
+def continuous_market(vega):
+    return setup_continuous_market(vega)
+
+def handle_route_connection_lost(route: Route, request):
+        if request.method == "POST" and re.match(r"http://localhost:\d+/api/v2/requests", request.url):
+            route.fulfill(
+                status=200,
+                headers={"Content-Type": "application/json"},
+                body='{"jsonrpc": "2.0", "id": "1"}'
+            )
+        else:
+            route.continue_()
+
+def handle_route_connection_rejected(route: Route, request):
+        if request.method == "POST" and re.match(r"http://localhost:\d+/api/v2/requests", request.url):
+            custom_response = {
+                "jsonrpc": "2.0",
+                "error": {
+                    "code": 3001,
+                    "data": "the user rejected the wallet connection",
+                    "message": "User error"
+                },
+                "id": "0"
+            }
+            route.fulfill(
+                status=400,
+                headers={"Content-Type": "application/json"},
+                body=json.dumps(custom_response)
+            )
+        else:
+            route.continue_()
+
+def assert_connection_approve(route: Route, request, page:Page):
+        if request.method == "POST" and re.match(r"http://localhost:\d+/api/v2/requests", request.url):
+            expect(page.get_by_test_id("toast-content")).to_have_text("Please go to your Vega wallet application and approve or reject the transaction.")
+        else:
+            route.continue_()
+
+@pytest.mark.usefixtures("page", "auth", "risk_accepted")
+def test_wallet_connection_error(continuous_market, page: Page):
+    page.goto(f"/#/markets/{continuous_market}")
+    page.route("**/*", handle_route_connection_lost)
+    page.get_by_test_id("connect-vega-wallet").click()
+    page.get_by_test_id("connector-jsonRpc").click()
+    expect(page.get_by_test_id("wallet-dialog-title")).to_have_text("Something went wrong")
+
+@pytest.mark.usefixtures("page", "risk_accepted")
+def test_wallet_connection_rejected(continuous_market, page: Page):
+    # 0002-WCON-002
+    # 0002-WCON-005
+    # 0002-WCON-007
+    # 0002-WCON-015
+    page.goto(f"/#/markets/{continuous_market}")
+    page.route("**/*", handle_route_connection_rejected)
+    page.get_by_test_id("connect-vega-wallet").click()
+    page.get_by_test_id("connector-jsonRpc").click()
+    expect(page.get_by_test_id("dialog-content").nth(1)).to_have_text("User errorthe user rejected the wallet connectionTry againAbout the Vega wallet  | Supported browsers ")
```

Comment:
oooh this expected text is odd... I guess it passes though? 

dialog-content, what does this element look like sorry? Only reason I say this is because locating things by nth child is not particularly robust in my experience and I would normally advise against it as a rule

Reply:
![image](https://github.com/vegaprotocol/frontend-monorepo/assets/133853298/3fa168fb-cd0d-4140-bc4f-19f158e5df39)
Basically the test id is duplicated I can look fixing the test ids but this will break quite a few tests so should maybe sit in a separate pr